test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetVerifyTests.cs

```diff
@@ -141,5 +144,508 @@ public void Verify_MultipleSignedPackagesWithWildCardAndDetailedVerbosity_MixedR
+                result.Success.Should().BeFalse();
+                result.AllOutput.Should().Contain(_noTimestamperWarningCode);
+                result.AllOutput.Should().Contain(_primarySignatureInvalidErrorCode);
+                result.AllOutput.Should().Contain("The repository primary signature's signing certificate is not trusted by the trust provider.");
+            }
+        }
+
+        [PlatformTheory(Platform.Windows, Platform.Linux)] // https://github.com/NuGet/Home/issues/11178
+        [InlineData("true", true)]
+        [InlineData("true", false)]
+        [InlineData("TRUE", true)]
+        [InlineData("TRUE", false)]
+        public void Verify_RepositorySignedPackageWithUntrustedCertificate_RepositoryTag_AllowUntrustedRootSetTrue_Succeeds(string trust, bool fingerPrintOption)
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var testDirectory = pathContext.WorkingDirectory;
+                // Signed with cert with untrusted root
+                var packageFile = new FileInfo(Path.Combine(testDirectory, "Test.Reposigned.1.0.0.nupkg"));
+                var package = SigningTestUtility.GetResourceBytes(packageFile.Name);
+                File.WriteAllBytes(packageFile.FullName, package);
+
+                var certificateFingerprintString = "775AAB607AA76028A7CC7A873A9513FF0C3B40DF09B7B83D21689A3675B34D9A";
+
+                var trustedSignersSectionContent = $@"
+    <trustedSigners>
+        <repository name=""MyCert"" serviceIndex=""{pathContext.PackageSource}"">
+            <certificate fingerprint=""{certificateFingerprintString}"" hashAlgorithm=""SHA256"" allowUntrustedRoot=""{trust}"" />
+        </repository>
+    </trustedSigners>
+";
+                SimpleTestSettingsContext.AddSectionIntoNuGetConfig(pathContext.WorkingDirectory, trustedSignersSectionContent, "configuration");
+                string fingerprint = fingerPrintOption ? $"--certificate-fingerprint {certificateFingerprintString}  --certificate-fingerprint DEF" : string.Empty;
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    testDirectory,
+                    $"nuget verify {packageFile.FullName} " +
+                    fingerprint,
+                    ignoreExitCode: true);
+
+                // Assert
+                // If allowUntrustedRoot is set true in nuget.config then verify succeeds for cert with untrusted root.
+                result.Success.Should().BeTrue();
+                result.AllOutput.Should().Contain(_noTimestamperWarningCode);
+            }
+        }
+
+        [PlatformTheory(Platform.Windows, Platform.Linux)] // https://github.com/NuGet/Home/issues/11178
+        [InlineData("true", true)]
+        [InlineData("false", true)]
+        [InlineData("true", false)]
+        [InlineData("false", false)]
+        [InlineData("TRUE", true)]
```

Comment:
Same feedback applies as in the previous scenario.

Reply:
Sure, removed.